### ISSUE TITLE
.keras_mnist.ipynb

### ISSUE THREAD

**Issue by Hank666818** [Hank666818/-#2]
from tensorflow import keras
from keras.models import Sequential
from keras.layers import Dense, Flatten
model=Sequential()
model.add(Flatten())
model.add(Dense(32, activation='relu', input_dim=784))
model.add(Dense(10, activation='sigmoid'))
model.compile(optimizer='rmsprop', loss='sparse_categorical_crossentropy', metrics=['accuracy'])
     

mnist=keras.datasets.mnist
(train_data, train_label), (test_data, test_label) = mnist.load_data()
     

print(type(train_data))
print(train_data.shape)
print(type(train_label))
print(train_label.shape)
print(test_data.shape)
print(test_label.shape)
     
<class 'numpy.ndarray'>
(60000, 28, 28)
<class 'numpy.ndarray'>
(60000,)
(10000, 28, 28)
(10000,)

print(train_label[0])
print(train_data[0])
     
5
[[  0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
    0   0   0   0   0   0   0   0   0   0]
 [  0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
    0   0   0   0   0   0   0   0   0   0]
 [  0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
    0   0   0   0   0   0   0   0   0   0]
 [  0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
    0   0   0   0   0   0   0   0   0   0]
 [  0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
    0   0   0   0   0   0   0   0   0   0]
 [  0   0   0   0   0   0   0   0   0   0   0   0   3  18  18  18 126 136
  175  26 166 255 247 127   0   0   0   0]
 [  0   0   0   0   0   0   0   0  30  36  94 154 170 253 253 253 253 253
  225 172 253 242 195  64   0   0   0   0]
 [  0   0   0   0   0   0   0  49 238 253 253 253 253 253 253 253 253 251
   93  82  82  56  39   0   0   0   0   0]
 [  0   0   0   0   0   0   0  18 219 253 253 253 253 253 198 182 247 241
    0   0   0   0   0   0   0   0   0   0]
 [  0   0   0   0   0   0   0   0  80 156 107 253 253 205  11   0  43 154
    0   0   0   0   0   0   0   0   0   0]
 [  0   0   0   0   0   0   0   0   0  14   1 154 253  90   0   0   0   0
    0   0   0   0   0   0   0   0   0   0]
 [  0   0   0   0   0   0   0   0   0   0   0 139 253 190   2   0   0   0
    0   0   0   0   0   0   0   0   0   0]
 [  0   0   0   0   0   0   0   0   0   0   0  11 190 253  70   0   0   0
    0   0   0   0   0   0   0   0   0   0]
 [  0   0   0   0   0   0   0   0   0   0   0   0  35 241 225 160 108   1
    0   0   0   0   0   0   0   0   0   0]
 [  0   0   0   0   0   0   0   0   0   0   0   0   0  81 240 253 253 119
   25   0   0   0   0   0   0   0   0   0]
 [  0   0   0   0   0   0   0   0   0   0   0   0   0   0  45 186 253 253
  150  27   0   0   0   0   0   0   0   0]
 [  0   0   0   0   0   0   0   0   0   0   0   0   0   0   0  16  93 252
  253 187   0   0   0   0   0   0   0   0]
 [  0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0 249
  253 249  64   0   0   0   0   0   0   0]
 [  0   0   0   0   0   0   0   0   0   0   0   0   0   0  46 130 183 253
  253 207   2   0   0   0   0   0   0   0]
 [  0   0   0   0   0   0   0   0   0   0   0   0  39 148 229 253 253 253
  250 182   0   0   0   0   0   0   0   0]
 [  0   0   0   0   0   0   0   0   0   0  24 114 221 253 253 253 253 201
   78   0   0   0   0   0   0   0   0   0]
 [  0   0   0   0   0   0   0   0  23  66 213 253 253 253 253 198  81   2
    0   0   0   0   0   0   0   0   0   0]
 [  0   0   0   0   0   0  18 171 219 253 253 253 253 195  80   9   0   0
    0   0   0   0   0   0   0   0   0   0]
 [  0   0   0   0  55 172 226 253 253 253 253 244 133  11   0   0   0   0
    0   0   0   0   0   0   0   0   0   0]
 [  0   0   0   0 136 253 253 253 212 135 132  16   0   0   0   0   0   0
    0   0   0   0   0   0   0   0   0   0]
 [  0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
    0   0   0   0   0   0   0   0   0   0]
 [  0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
    0   0   0   0   0   0   0   0   0   0]
 [  0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
    0   0   0   0   0   0   0   0   0   0]]

import matplotlib.pyplot as plt
plt.imshow(train_data[0], cmap='binary')
plt.show()
     

model.fit(train_data,train_label,epochs=20,batch_size=512)
     
Epoch 1/20
118/118 [==============================] - 2s 10ms/step - loss: 5.5679 - accuracy: 0.4254
Epoch 2/20
118/118 [==============================] - 1s 10ms/step - loss: 1.1698 - accuracy: 0.6519
Epoch 3/20
118/118 [==============================] - 1s 10ms/step - loss: 0.8294 - accuracy: 0.8076
Epoch 4/20
118/118 [==============================] - 1s 10ms/step - loss: 0.6627 - accuracy: 0.8563
Epoch 5/20
118/118 [==============================] - 1s 11ms/step - loss: 0.5409 - accuracy: 0.8754
Epoch 6/20
118/118 [==============================] - 1s 11ms/step - loss: 0.4570 - accuracy: 0.8899
Epoch 7/20
118/118 [==============================] - 1s 10ms/step - loss: 0.3965 - accuracy: 0.9010
Epoch 8/20
118/118 [==============================] - 1s 11ms/step - loss: 0.3584 - accuracy: 0.9097
Epoch 9/20
118/118 [==============================] - 1s 11ms/step - loss: 0.3207 - accuracy: 0.9178
Epoch 10/20
118/118 [==============================] - 1s 12ms/step - loss: 0.2997 - accuracy: 0.9213
Epoch 11/20
118/118 [==============================] - 1s 11ms/step - loss: 0.2780 - accuracy: 0.9266
Epoch 12/20
118/118 [==============================] - 1s 8ms/step - loss: 0.2620 - accuracy: 0.9306
Epoch 13/20
118/118 [==============================] - 1s 7ms/step - loss: 0.2480 - accuracy: 0.9320
Epoch 14/20
118/118 [==============================] - 1s 6ms/step - loss: 0.2323 - accuracy: 0.9351
Epoch 15/20
118/118 [==============================] - 1s 6ms/step - loss: 0.2208 - accuracy: 0.9385
Epoch 16/20
118/118 [==============================] - 1s 6ms/step - loss: 0.2121 - accuracy: 0.9396
Epoch 17/20
118/118 [==============================] - 1s 6ms/step - loss: 0.2005 - accuracy: 0.9435
Epoch 18/20
118/118 [==============================] - 1s 6ms/step - loss: 0.1918 - accuracy: 0.9441
Epoch 19/20
118/118 [==============================] - 1s 7ms/step - loss: 0.1852 - accuracy: 0.9467
Epoch 20/20
118/118 [==============================] - 1s 6ms/step - loss: 0.1776 - accuracy: 0.9488
<keras.callbacks.History at 0x7f25f5b49790>

score=model.evaluate(test_data, test_label)
print(score[1])
     
313/313 [==============================] - 1s 2ms/step - loss: 0.3632 - accuracy: 0.9305
0.9304999709129333
